pyanaconda/modules/payloads/payload/dnf/initialization.py

```diff
@@ -35,3 +51,243 @@ def configure_dnf_logging():
     # Set up DNF. Increase the log level to the custom DDEBUG level.
     dnf_logger = logging.getLogger(DNF_LOGGER)
     dnf_logger.setLevel(dnf.logging.DDEBUG)
+
+
+SetUpDNFSourcesResult = namedtuple(
+    "LoadTreeInfoMetadataResult", [
+        "dnf_manager",
+        "repositories",
+        "sources",
+    ]
+)
+SetUpDNFSourcesResult.__doc__ += """
+The result of the SetUpDNFSourcesTask task.
+"""
+
+
+class SetUpDNFSourcesTask(SetUpSourcesTask):
+    """Set up all the installation source of the DNF payload."""
+
+    def __init__(self, sources, repositories, configuration):
+        """Create a new task."""
+        super().__init__(sources)
+        self._configuration = configuration
+        self._repositories = repositories
+        self._treeinfo_repositories = []
+        self._release_version = None
+        self._proxy = None
+
+    @property
+    def _source(self):
+        return self._sources[0]
+
+    def run(self):
+        """Run the task.
+
+        The task returns a named tuple with the following values:
+
+            dnf_manager     a DNF manager with a configured base
+            repositories    an updated list of additional repositories
+            sources         a list of generated additional sources
+
+        :return SetUpDNFSourcesResult: a result of the task
+        """
+        # Set up the main source.
+        super().run()
+
+        # Process the configuration of the main source.
+        repository = self._process_source_metadata(self._source)
+
+        # Process the treeinfo metadata of the main source.
+        # Update the treeinfo repositories if any.
+        repositories = update_treeinfo_repositories(
+            self._repositories,
+            self._treeinfo_repositories
+        )
+
+        # Configure the DNF manager.
+        dnf_manager = self._configure_dnf_manager()
+
+        # Generate and set up additional sources.
+        sources = self._generate_additional_sources(
+            repositories=repositories,
+            substitute=dnf_manager.substitute
+        )
+
+        self._set_up_sources(sources)
+
+        # Load the main source.
+        self._load_source(dnf_manager, self._source, repository)
+
+        # Load additional sources.
+        self._load_additional_sources(dnf_manager, sources, repositories)
+
+        # Validate enabled repositories.
+        self._validate_repositories(dnf_manager)
+
+        # Load package and group metadata.
+        self.report_progress(_("Downloading group metadata..."))
+        self._load_metadata(dnf_manager)
```

Comment:
This is a minor thing, just wondering. The single `report_progress` seems weird. Why does the text disagree with the comment and code? And the other steps are not reported? What's the initial state, empty string? Or is this actually secretly intertwined with GUI where the rest of the messages come from?

Reply:
This message comes from the previous code and I didn't wanted to change the statuses visible in GUI too much. However, this can be easily changed if it is ever an issue. And yes, some of the other messages are still in the ui-related part of the payload code.

Reply:
Thanks, I thought it's to preserve the messages 1:1 with the previous state.